### PR TITLE
Renames CustomPiwikJs plugin to CustomJsTracker

### LINKED ISSUE #13604: Rename CustomPiwikJs plugin to CustomMatomoJs (or CustomTrackerJs)
Follow up issue from  https://github.com/matomo-org/matomo/pull/13596 and https://github.com/matomo-org/matomo/issues/12785

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #13604

### CODE REVIEW

**Comment by tsteur** [comment]
@sgiehl there used to be a plugin on the marketplace named `CustomTrackerJs`... just wondering if someone updates from Piwik 2.X to Matomo 4.X if this could cause any issues? I suppose that plugin would be simply overwritten? (which be kind of good since it would also automatically remove an outdated / not compatbile plugin) or is easier to rename it to `CustomJsTracker`?

**Comment by sgiehl** [comment]
Guess it would be overwritten, but not sure if some old files might remain 🤷‍♂ 
Not having a preference here. Can also rename it...

**Comment by tsteur** [comment]
Up to you. If we keep the name, might be worth testing what happens when upgrading from Piwik 2 and such a plugin existed. Might be easier to maybe rename it?

**Comment by sgiehl** [comment]
renamed it again. Need to update the UI tests again, I guess. But should be ready for another review

**Review by tsteur — CHANGES_REQUESTED** [review]
Left a few minor comments, nothing major. Mostly around docs.

Adjusted already 6 or so internal plugins. Be good to make the changes in tag manager and resolve the open issues then be good to merge

File: core/Updates/4.0.0-b1.php [hunk 1]
```diff
@@ -45,5 +47,31 @@ public function getMigrations(Updater $updater)
     public function doUpdate(Updater $updater)
     {
         $updater->executeMigrations(__FILE__, $this->getMigrations($updater));
+
+        $this->renameTrackerJsPluginInConfig();
```
[the remark this answers is not in the document]
Comment (sgiehl) [reply core/Updates/4.0.0-b1.php]: I'll adjust the old update script to activate `CustomJsTracker` instead. That way it should be activated if someone updates from an older version

File: core/Updates/4.0.0-b1.php [hunk 2]
```diff
@@ -45,5 +47,31 @@ public function getMigrations(Updater $updater)
...
+
+        Cache::flushAll();
+    }
+
+    protected function renameTrackerJsPluginInConfig()
+    {
+        $updater = new Updater();
+        $updater->markComponentSuccessfullyUninstalled('CustomPiwikJs');
+
+        $config = Config::getInstance();
```
[the remark this answers is not in the document]
Comment (sgiehl) [reply core/Updates/4.0.0-b1.php]: Not that easy. The plugin `CustomPiwikJS ` won't be reported as active as soon as it is missing on file system. But an `uninstall` should work. Will build a migration for that

**Comment by sgiehl** [comment]
@tsteur applied the feedback

**Comment by tsteur** [comment]
Looks good in general @sgiehl some tests will need fixing though

**Comment by sgiehl** [comment]
Guess tests are failing because the changes in tagmanager are needed. will push a submodule change to prove that.

**Comment by sgiehl** [comment]
Tests seem to pass again. Before merging we need to merge https://github.com/matomo-org/tag-manager/pull/220 and update the submodule ref here.

**Pull request merged**